starcheck/plot.py

```diff
@@ -0,0 +1,187 @@
+               s=symsize(np.array([10.0, 9.0, 8.0, 7.0, 6.0])))
+
+    [l.set_rotation(90) for l in ax.get_yticklabels()]
+    ax.grid()
+    ax.set_ylabel("Zag (arcsec)")
+    ax.set_xlabel("Yag (arcsec)")
+
+    # plot starcheck catalog
+    if catalog is not None:
+        cat = Table(catalog)
+        gui = cat[cat['type'] == 'GUI']
+        acq = cat[cat['type'] == 'ACQ']
+        bot = cat[cat['type'] == 'BOT']
+        fid = cat[cat['type'] == 'FID']
+        for row in cat:
+            ax.annotate("%s" % row['idx'],
+                        xy=(row['yang'] - 120, row['zang'] + 60))
+        ax.scatter(gui['yang'], gui['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   s=100)
+        ax.scatter(bot['yang'], bot['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   s=100)
+        for acq_star in acq:
+            acq_box = plt.Rectangle(
+                (acq_star['yang'] - acq_star['halfw'],
+                 acq_star['zang'] - acq_star['halfw']),
+                width=acq_star['halfw'] * 2,
+                height=acq_star['halfw'] * 2,
+                color='blue',
+                fill=False)
+            ax.add_patch(acq_box)
+        for acq_star in bot:
+            acq_box = plt.Rectangle(
+                (acq_star['yang'] - acq_star['halfw'],
+                 acq_star['zang'] - acq_star['halfw']),
+                width=acq_star['halfw'] * 2,
+                height=acq_star['halfw'] * 2,
+                color='blue',
+                fill=False)
+            ax.add_patch(acq_box)
+        ax.scatter(fid['yang'], fid['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   marker='o',
+                   s=200)
+        ax.scatter(fid['yang'], fid['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   marker='+',
+                   linewidth=3,
+                   s=200)
+
+    # plot field if present
+    if field is not None:
+        faint_plot_mag = 11.0
+        bright_field = field[field['MAG_ACA'] < faint_plot_mag]
+        yags = []
+        zags = []
+        for star in bright_field:
+            yag, zag = radec2yagzag(star['RA_PMCORR'],
+                                    star['DEC_PMCORR'],
+                                    quat)
+            yag *= 3600
+            zag *= 3600
+            yags.append(yag)
+            zags.append(zag)
+
+        bright_field = Ska.Numpy.add_column(bright_field,
+                                            'yang',
+                                            yags)
+        bright_field = Ska.Numpy.add_column(bright_field,
+                                            'zang',
+                                            zags)
+        color = np.ones(len(bright_field), dtype='|S10')
+        color[:] = 'red'
+        faint = ((bright_field['CLASS'] == 0)
```

Comment:
Should it be a kwarg or should I make field an object that has that property or some such?  Not sure which might make more sense for interface.